io-engine/src/core/volume_snapshot.rs

```diff
@@ -0,0 +1,37 @@
+use super::CoreError;
+use crate::{lvs::LvolSnapshot, subsys::NvmfReq};
+use async_trait::async_trait;
+use spdk_rs::libspdk::spdk_bdev_io;
+
+///  Traits gives the common snapshot/clone interface for Local/Remote Lvol.
+#[async_trait(?Send)]
+pub trait VolumeSnapshotOps {
```

Comment:
Seems like we could split this trait into 2?
Example we want to provide the "getters" here for snapshot, and they don't need to be an option then?
And we don't wan the snapshot itself to impl create_snapshot?